app/[locale]/(technician)/farmer/[id]/(dashboard)/layout.tsx

```diff
@@ -30,39 +32,46 @@ export async function getFarmerName(id: string) {
   }
 }
 
-export async function generateMetadata({ params }: { params: { id: string } }, parent?: ResolvingMetadata) {
-  const name = await getFarmerName(params.id)
+export async function generateMetadata({ params }: { params: { locale: string } }) {
+  const t = await getTranslator(params.locale, "FarmerDetail")
   return {
-    title: `${name || "Farmer Detail"} | Ovoflow`
+    title: t("title")
   } as Metadata
 }
 
 export default async function FarmerDetail({ profile, params, summary, visits }: Props) {
-  const now = Date.now()
-  const lastWeek = now - 7 * 24 * 3600 * 1000;
+  const locale = useLocale()
+  const t = await getTranslator(locale, "FarmerDetail")
+  const now = await getNow(locale)
+  const lastWeek = +now - 7 * 24 * 3600 * 1000;
   const name = await getFarmerName(params.id)
+  const formatter = new Intl.DateTimeFormat(locale, {
+    month: "long",
+    day: "numeric"
+  })
```

Comment:
e.g.

Reply:
However, the `formatter` doesn't have the format range support (see line 63 below). That's why I need a new `Intl.DateTimeFormat` object